integration/apps/hpl_cpu/hpl_cpu.py

```diff
@@ -0,0 +1,155 @@
+#        distribution.
+#
+#      * Neither the name of Intel Corporation nor the names of its
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+import os
+import sys
+import glob
+import math
+import textwrap
+
+from apps import apps
+
+
+class HplCpuAppConf(apps.AppConf):
+    @staticmethod
+    def name():
+        return 'hpl_cpu'
+
+    def __init__(self, num_nodes, mach, perc_dram_per_node=0.9, cores_per_node=None):
+        '''
+        num_nodes: Number of MPI ranks (1 node per rank) -- 2, 4, 8 or 16.
+        perc_dram_per_node: Ratio of the total node DRAM that should be used for the
+                            HPL matrix (assuming DP).
+                            80-90% is a good amount to maximize efficiency.
+                            Default is 0.9.
+        cores_per_node: Number of Xeon cores that each MPI process can offload to via OMP.
+                        Total number of physical cores will be selected if this is not set
+                        (defauilt=None).
+        '''
+        dram_for_app = num_nodes * mach.total_node_memory_bytes() * perc_dram_per_node
+        if cores_per_node is None:
+            cores_per_node = mach.num_core()
+
+        benchmark_dir = os.path.dirname(os.path.abspath(__file__))
+        self.exe_path = os.path.join(benchmark_dir, 'hpl-2.3/bin/Linux_Intel64/xhpl')
+        self.NBs=384 # This is the recommended size for Intel Scalable Xeon family.
+        process_grid_ratios = {
+            1: {'P': 1, 'Q': 1},
+            2: {'P': 1, 'Q': 2},
+            4: {'P': 2, 'Q': 2},
+            8: {'P': 2, 'Q': 4},
+            16: {'P': 4, 'Q': 4}
+        }
+        if num_nodes not in process_grid_ratios:
+            raise RuntimeError(f"Number of nodes {num_nodes} is not defined for HPL.")
+        self.P = process_grid_ratios[num_nodes]['P']
+        self.Q = process_grid_ratios[num_nodes]['Q']
+
+        self.N = int(round(math.sqrt(dram_for_app / 8)))
+        self._cpu_per_rank = cores_per_node
+
+        print(f'DRAM reserved for APP: {dram_for_app} Cores for app: {cores_per_node} N={self.N}')
```

Comment:
Please use sys.stdout.write() if you need to print to stdout.  print() is deprecated.

Reply:
Wow this is news to me :) I thought with Python3 "print something" was replaced with "print(something)" (statement vs function). I could not find any reference to print function to be deprecated though. When did that happen?

Reply:
You're correct about the print statement vs function.  I misspoke a bit about it being deprecated.

We never use print (statement or function) in any of the Python code in the GEOPM repo.  We still have to support Python 2, and sys.stdout.write() works the same in Python 2 and 3.

Reply:
No problem, I will change it. However I should also mention that I was not able to run the integration scripts with Python 2 (yaml packages missing, etc). Maybe it is because I don't have my Python 2 packages up to date?